Unexpected ToleranceMatcher behavior at infinite doubles

**Which version of Kotest are you using**
4.3.1

Mathematically I would expect

    checkAll(Arb.positiveDoubles()) { eps ->
      Double.NEGATIVE_INFINITY shouldBe (Double.NEGATIVE_INFINITY plusOrMinus eps)
      Double.POSITIVE_INFINITY shouldBe (Double.POSITIVE_INFINITY plusOrMinus eps)
    }

to succeed. But Kotest fails with

    Property test failed for inputs

    0) 1.0

    Caused by java.lang.AssertionError: -Infinity should be equal to -Infinity within tolerance of 1.0 (lowest acceptable value is -Infinity; highest acceptable value is -Infinity) at
	[...]
	io.kotest.property.internal.ProptestKt$proptest$$inlined$forEach$lambda$1.invokeSuspend(proptest.kt:32)
	io.kotest.property.internal.ProptestKt$proptest$$inlined$forEach$lambda$1.invoke(proptest.kt)

--- a/kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/doubles/Tolerance.kt
+++ b/kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/doubles/Tolerance.kt
@@ -14,14 +14,14 @@ import kotlin.math.abs
  * ```
  */
 infix fun Double.plusOrMinus(tolerance: Double): ToleranceMatcher {
-   require(tolerance >= 0)
+   require(tolerance >= 0 && tolerance.isFinite())
    return ToleranceMatcher(this, tolerance)
 }
 
 class ToleranceMatcher(private val expected: Double?, private val tolerance: Double) : Matcher<Double?> {
 
   override fun test(value: Double?): MatcherResult {
-    return if(value == null || expected == null) {
+    return if(value == null || expected == null || expected.isInfinite()) {
       MatcherResult(value == expected, "$value should be equal to $expected", "$value should not be equal to $expected")
     } else if (expected.isNaN() && value.isNaN()) {
        println("[WARN] By design, Double.Nan != Double.Nan; see https://stackoverflow.com/questions/8819738/why-does-double-nan-double-nan-return-false/8819776#8819776")

--- a/kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/doubles/DoubleToleranceTest.kt
+++ b/kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/doubles/DoubleToleranceTest.kt
@@ -5,7 +5,7 @@ import io.kotest.core.spec.style.FunSpec
 import io.kotest.matchers.doubles.plusOrMinus
 import io.kotest.matchers.shouldBe
 
-class DoubleTolerenceTest : FunSpec({
+class DoubleToleranceTest : FunSpec({
 
    test("double with tolerance should include tolerance in error message") {
       shouldThrowAny {

--- a/kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/doubles/DoubleToleranceTest.kt
+++ b/kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/doubles/DoubleToleranceTest.kt
@@ -4,6 +4,9 @@ import io.kotest.assertions.throwables.shouldThrowAny
 import io.kotest.core.spec.style.FunSpec
 import io.kotest.matchers.doubles.plusOrMinus
 import io.kotest.matchers.shouldBe
+import io.kotest.property.Arb
+import io.kotest.property.arbitrary.numericDoubles
+import io.kotest.property.checkAll
 
 class DoubleToleranceTest : FunSpec({
 
@@ -11,6 +14,13 @@ class DoubleToleranceTest : FunSpec({
       shouldThrowAny {
          1.0 shouldBe (1.5 plusOrMinus 0.4)
       }.message shouldBe "1.0 should be equal to 1.5 within tolerance of 0.4 (lowest acceptable value is 1.1; highest acceptable value is 1.9)"
+   }
+
+   test("infinite double with finite tolerance should equal the same infinite double") {
+      checkAll(Arb.numericDoubles(from = 0.0)) { eps ->
+         Double.NEGATIVE_INFINITY shouldBe (Double.NEGATIVE_INFINITY plusOrMinus eps)
+         Double.POSITIVE_INFINITY shouldBe (Double.POSITIVE_INFINITY plusOrMinus eps)
+      }
    }
 })
 

